feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts

```diff
@@ -0,0 +1,158 @@
+import { ConfiguratorAttributeBaseComponent } from './configurator-attribute-base.component';
+import { Directive, EventEmitter, Input, Output } from '@angular/core';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorAttributeQuantityComponentOptions } from '../../quantity/configurator-attribute-quantity.component';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
+
+@Directive()
+export abstract class ConfiguratorAttributeSingleSelectionBaseComponent extends ConfiguratorAttributeBaseComponent {
+  loading$ = new BehaviorSubject<boolean>(false);
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() ownerKey: string;
+  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  // TODO(#11681): make quantityService a required dependency
+  /**
+   * default constructor
+   * @param {ConfiguratorAttributeQuantityService} quantityService
+   */
+  // eslint-disable-next-line @typescript-eslint/unified-signatures
+  constructor(quantityService: ConfiguratorAttributeQuantityService);
+
+  /**
+   * @deprecated since 3.3
+   */
+  constructor();
+
+  constructor(
+    protected quantityService?: ConfiguratorAttributeQuantityService
+  ) {
+    super();
+  }
+
+  /**
+   * Checks if we are supposed to render a quantity control, which
+   * can be derived from the attribute meta data
+   *
+   * @return {boolean} - Display quantity picker?
+   */
+  get withQuantity(): boolean {
+    return (
+      this.quantityService?.withQuantity(
+        this.attribute?.dataType ?? Configurator.DataType.NOT_IMPLEMENTED,
+        this.attribute?.uiType ?? Configurator.UiType.NOT_IMPLEMENTED
+      ) ?? false
+    );
+  }
+
+  /**
+   * Checks if quantity control should be disabled
+   *
+   * @return {boolean} - Disable quantity picker?
+   */
+  get disableQuantityActions(): boolean {
+    return (
+      this.quantityService?.disableQuantityActions(
+        this.attribute.selectedSingleValue
+      ) ?? true
+    );
+  }
+
+  onSelect(value: string): void {
```

Comment:
Please create a service, and move `onSelect`, `onDeselect`, `onHandleQuantity`, `onChangeQuantity`, `extractQuantityParameters`, `extractPriceFormulaParameters` to there. So, in child component, instead of call `super.xxx`, we can just call `service.xxx`.